pages/implementation/reference/javascript.mdx

```diff
@@ -0,0 +1,201 @@
+
+and you call it like this
+
+```javascript
+Avo.initAvo(
+  { env: 'dev', noop: false, strict: false, avoLogger: avoLogger, inspector: avoInspector },
+  { ...systemProperties },
+  { appsFlyer: { appId: 'XXXXXX' } },
+  customDestination
+);
+```
+
+#### About the custom destinations
+
+The last parameter in `initAvo` is special. If you choose to send your tracking to a custom destination in the Avo UI, it will be generated.
+Custom destination is a callback that Avo calls after all the checks. It provides the data ready to be sent to any analytics destination.
+
+The callback methods mirror the actions you set up for given event in your tracking plan
+
+<center style={{marginBottom: 16 + "px"}}>
+  <img
+    src={require('../../../images/event-actions.png')}
+    alt="List of event actions in Avo: Log Event, Update User Properties, Identify User, Unidentify User, Log Revenue, Log Page View"
+  />
+</center>
+
+```javascript
+import Avo from './Avo';
+
+// An object representing your custom destination
+// that you pass into initAvo():
+
+var customDestination = {
+  make: function (env) {
+    // called at the end of the initAvo call
+    // You should initialize your analytics destination here, if you have not doen it before
+    analytics.load(apiKey, options);
+  },
+  logEvent: function (eventName, eventProperties) {
+    // called when you invoke Avo functions with the Log Event action
+    analytics.track(eventName, eventProperties);
+  },
+  setUserProperties: function (userId, userProperties) {
+   // called when you invoke Avo functions with the Update User Propreties action
+   analytics.identify(userProperties);
+  },
+  identify: function (userId) {
+    // called when you invoke Avo functions with the Identify User action
+    analytics.identify(userId, {});
+  },
+  unidentify: function () {
+    // called when you invoke Avo functions with the Unidentify User action
+    analytics.identify(null, {});
+  },
+  revenue: function(amount, eventProperties) {
+    // called when you invoke Avo functions with the Log Revenue action
+    analytics.track("revenue", {"revenue": amount});
+  },
+  page: function(pageName, eventProperties) {
+    // called when you invoke Avo functions with the Log Page View action
+    analytics.page(eventName, eventProperties);
+  }
+};
+```
+
+#### Step 3. Call Avo functions to track your product usage
+
+Every event in your tracking plan, marked with the "Implement with Avo" checkmark, gets a function in the generated code, name according to the event name, in lower camel case.
+
+For example, if you have an event defined like this in Avo
+
+<center style={{marginBottom: 16 + "px"}}>
+  <img
+    src={require('../../../images/signup-start-ui.png')}
+    alt="Event 'Signup Start' defined in Avo with referral string property and implement with Avo check"
+  />
+</center>
+
+You'll be able to call it like this from the generated code
+
+```javascript
+Avo.signupStart({ referral: "direct" });
+```
```

Comment:
Should we use the same lodash pattern here, where we have the conventional function definition, followed by an example?

```
Avo.signupStart(properties)
```
Arguments:
- properties: ...

Example:
...